src/components/BookDetails.jsx

```diff
@@ -1,9 +1,38 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import '../css/BookDetails.css';
 import QrReader from './QrReader';
+import { fetchMyTransactions, fetchTransactionData, isTaken } from '../helpers/apiRequests';
 
-const BookDetails = ({ book,updateBooks}) => {
+const BookDetails = ({ book }) => {
+  const [buttonText, setButtonText] = useState(null);
 
+  useEffect(() => {
+    console.log('Book details:', book);
+    checkStatus(); 
+  }, []);
+  
+  const checkStatus = async () => {
+    try {
+      const transactions = await fetchMyTransactions();
+      console.log('Transactions:', transactions);

```

Comment:
pls delete that console.log it contains test data

Reply:
deleting the log